accounts/src/inmemory_account_store.rs

```diff
@@ -74,12 +78,14 @@ impl AccountDataByCommitment {
             .map(|x| x.updated_slot < data.updated_slot)
             .unwrap_or(true);
 
+        let mut updated = false;
         if self.processed_accounts.get(&data.updated_slot).is_none() {
             // processed not present for the slot
             self.processed_accounts
                 .insert(data.updated_slot, data.clone());
```

Comment:
this misses the case in this flow:

GIVEN emptyMap, None, None
WHEN
 `.update(1234, confirmed)` for a slot that was not registered before
THEN_WRONG
 map(1234, None, None)
THEN_CORRECT
 map(1234, 1234, None)